File: 1342-number-of-steps-to-reduce-a-number-to-zero/1342-number-of-steps-to-reduce-a-number-to-zero.js
/**
 * @param {number} num
 * @return {number}
 */
var numberOfSteps = function(num) {
    let  count = 0
    while(num>0){
        if(num&1 == 1){
           num--
           count++  
           console.log(num)
           continue
        } 
        num =  num >> 1
        console.log(num)
        count++
    }
    return count
};

File: 1342-number-of-steps-to-reduce-a-number-to-zero/1342-number-of-steps-to-reduce-a-number-to-zero.java
class Solution {
    public int numberOfSteps(int num) {
        int count =0;
        while(num>0){
            if((num&1)== 1){
                num--;
                count++;
                continue;
            }
            num = num >> 1;
            count++;
        }
        return count;
    }
}